beacon-chain/rpc/beacon_chain_server.go

```diff
@@ -17,32 +18,56 @@ import (
 // beacon chain.
 type BeaconChainServer struct {
 	beaconDB *db.BeaconDB
+	pool     operations.Pool
 }
 
 // ListAttestations retrieves attestations by block root, slot, or epoch.
 //
 // The server may return an empty list when no attestations match the given
 // filter criteria. This RPC should not return NOT_FOUND. Only one filter
 // criteria should be used.
+//
+// TODO(#3064): Filtering blocked by DB refactor for easier access to
+// fetching data by attributes efficiently.
 func (bs *BeaconChainServer) ListAttestations(
 	ctx context.Context, req *ethpb.ListAttestationsRequest,
 ) (*ethpb.ListAttestationsResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	switch req.QueryFilter.(type) {
+	case *ethpb.ListAttestationsRequest_BlockRoot:
+		return nil, status.Error(codes.Unimplemented, "not implemented")
+	case *ethpb.ListAttestationsRequest_Slot:
+		return nil, status.Error(codes.Unimplemented, "not implemented")
+	case *ethpb.ListAttestationsRequest_Epoch:
+		return nil, status.Error(codes.Unimplemented, "not implemented")
+	}
+	atts, err := bs.beaconDB.Attestations()
```

Comment:
I think we should add pagination for this, this could return a lot of attestations(over thousands)
@rauljordan @prestonvanloon

Reply:
Agreed!